workloads/imagenet/imagenet_jax/workload.py

```diff
@@ -0,0 +1,290 @@
+
+  _InitState = Tuple[spec.ParameterTree, spec.ModelAuxillaryState]
+  def init_model_fn(
+      self,
+      rng: spec.RandomState) -> _InitState:
+    model_cls = getattr(models, self.model_name)
+    model = self.create_model(model_cls=model_cls)
+    self._model = model
+    params, model_state = self.initialized(rng, model)
+    self._param_shapes = jax.tree_map(
+      lambda x: spec.ShapeTuple(x.shape),
+      params)
+    model_state = jax_utils.replicate(model_state)
+    params = jax_utils.replicate(params)
+    return params, model_state
+
+  def eval_model_fn(self, params, batch, state, rng):
+    logits, _ = self.model_fn(
+      params,
+      batch,
+      state,
+      spec.ForwardPassMode.EVAL,
+      rng,
+      update_batch_norm=False)
+    return self.compute_metrics(logits, batch['label'])
+
+  def model_fn(
+      self,
+      params: spec.ParameterTree,
+      augmented_and_preprocessed_input_batch: spec.Tensor,
+      model_state: spec.ModelAuxillaryState,
+      mode: spec.ForwardPassMode,
+      rng: spec.RandomState,
+      update_batch_norm: bool) -> Tuple[spec.Tensor, spec.ModelAuxillaryState]:
+    variables = {'params': params, **model_state}
+    train = mode == spec.ForwardPassMode.TRAIN
+    if update_batch_norm:
+      logits, new_model_state = self._model.apply(
+        variables,
+        jax.numpy.squeeze(augmented_and_preprocessed_input_batch['image']),
+        train=train,
+        mutable=['batch_stats'])
+      return logits, new_model_state
+    else:
+      logits = self._model.apply(
+        variables,
+        jax.numpy.squeeze(augmented_and_preprocessed_input_batch['image']),
+        train=train,
+        mutable=False)
+      return logits, None
+
+  # Does NOT apply regularization, which is left to the submitter to do in
+  # `update_params`.
+  def loss_fn(
+      self,
+      label_batch: spec.Tensor,
+      logits_batch: spec.Tensor) -> spec.Tensor:  # differentiable
+    """Cross Entropy Loss"""
+    one_hot_labels = common_utils.onehot(label_batch,
+                                         num_classes=self.num_classes)
+    xentropy = optax.softmax_cross_entropy(logits=logits_batch,
+                                           labels=one_hot_labels)
+    return jnp.mean(xentropy)
+
+  def compute_metrics(self, logits, labels):
+    loss = self.loss_fn(labels, logits)
+    accuracy = jnp.mean(jnp.argmax(logits, -1) == labels)
+    metrics = {
+      'loss': loss,
+      'accuracy': accuracy,
+    }
+    metrics = lax.pmean(metrics, axis_name='batch')
+    return metrics
+
+  def eval_model(
+      self,
+      params: spec.ParameterTree,
+      model_state: spec.ModelAuxillaryState,
+      rng: spec.RandomState,
+      data_dir: str):
+    """Run a full evaluation of the model."""
+
+    # sync batch statistics across replicas once per epoch
+    model_state = self.sync_batch_stats(model_state)
+
+    epoch_metrics = common_utils.get_metrics(self.epoch_metrics)
```

Comment:
for now, lets remove all of `epoch_metrics` (here, in `submission.py`, and in `__init__()`), we will need to add something like this later, but I would prefer to make a more universal solution for all workloads so we can have a unified logging pipeline (we'll need consistent logging code and file format so submitters can audit their submission performance if we run it).